core/client/src/main/java/alluxio/client/block/BlockStoreContext.java

```diff
@@ -159,14 +141,13 @@ public BlockWorkerClient acquireWorkerClient(String hostname) throws IOException
    * @return a {@link BlockWorkerClient} connected to the worker with the given hostname
    * @throws IOException if no Alluxio worker is available for the given hostname
    */
-  public BlockWorkerClient acquireWorkerClient(WorkerNetAddress address)
-      throws IOException {
+  public BlockWorkerClient acquireWorkerClient(WorkerNetAddress address) throws IOException {
     BlockWorkerClient client;
     if (address == null) {
       throw new RuntimeException(ExceptionMessage.NO_WORKER_AVAILABLE.getMessage());
     }
     if (address.getHost().equals(NetworkAddressUtils.getLocalHostName(ClientContext.getConf()))) {
-      client = acquireLocalWorkerClient();
+      client = acquireLocalWorkerClient(address);
```

Comment:
Would the following exception message be misleading if the address has the local hostname but the port does not exist as a registered worker?